src/python/analyzer_executor/src/analyzer_executor_lib/analyzer_executor.py

```diff
@@ -54,157 +53,331 @@ def get(self, key):
 
 EitherCache = Union[NopCache, redis.Redis]
 
-if IS_LOCAL:
-    message_cache: EitherCache = NopCache()
-    hit_cache: EitherCache = NopCache()
-else:
-    MESSAGECACHE_ADDR = os.environ["MESSAGECACHE_ADDR"]
-    MESSAGECACHE_PORT = int(os.environ["MESSAGECACHE_PORT"])
 
-    HITCACHE_ADDR = os.environ["HITCACHE_ADDR"]
-    HITCACHE_PORT = os.environ["HITCACHE_PORT"]
+class AnalyzerExecutor:
 
-    message_cache = redis.Redis(host=MESSAGECACHE_ADDR, port=MESSAGECACHE_PORT, db=0)
-    hit_cache = redis.Redis(host=HITCACHE_ADDR, port=int(HITCACHE_PORT), db=0)
+    # constants
+    CHUNK_SIZE_RETRY: int = 10
+    CHUNK_SIZE_DEFAULT: int = 100
 
+    # singleton
+    _singleton = None
```

Comment:
Could we rip out the singleton-ness of this? No real need for it.

- rename `def singleton` to `def create`
- remove the `_singleton` stuff everywhere
- each test case creates its own instance

Reply:
The singleton() class method is handy because it condenses all the weird statefulness (pulling from environment, etc.) into one method.  In that sense, it's supposed to be objectionable 😛

This strongly resembles the existing behavior of the flat file.  It moves eager init (everything upon file load) to lazy init (everything upon first singleton() call) while offering dependency injection for testability.  I'm completely on board with improving the design here, though I don't want to go down a garden path of design changes when I'm just trying to get Redis in local.

Reply:
I completely agree with the core argument that "this one method is handy because it condenses all the weird statefullness".

I'm saying that, with the singleton, we are indeed introducing a _new_ state - **has this been initialized already?** - that we can completely avoid by, instead, creating a new instance of AnalyzerExecutor for every test.

Reply:
I agree with you but I think that's scope creeping past this ticket.  The flat file approach was effectively a singleton; I'm not of the opinion that a singleton is preferable, but _for the core purpose of increasing test coverage and making dev resemble prod_ I want to keep collateral changes scoped down to the bare minimum for implementing tests.